crates/ra_syntax/src/lexer.rs

```diff
@@ -190,13 +190,12 @@ fn next_token_inner(c: char, ptr: &mut Ptr) -> SyntaxKind {
 }
 
 fn scan_ident(c: char, ptr: &mut Ptr) -> SyntaxKind {
-    let is_single_letter = match ptr.current() {
-        None => true,
-        Some(c) if !is_ident_continue(c) => true,
-        _ => false,
-    };
-    if is_single_letter {
-        return if c == '_' { UNDERSCORE } else { IDENT };
+    match (c, ptr.current()) {
+        ('r', Some('#')) => {
+            ptr.bump();
+        }
+        ('_', Some(c)) if !is_ident_continue(c) => return UNDERSCORE,
+        _ => {}
     }
     ptr.bump_while(is_ident_continue);
     if let Some(kind) = SyntaxKind::from_keyword(ptr.current_token_text()) {
```

Comment:
This shouldn't be used for raw idents.

Reply:
Oups, indeed, great catch!

I think we need a test like `fn foo() { let r#struct = 92; let r#trait = #struct * 2 }`

Reply:
True, but there's never going to be a keyword matching `r#<anything>`, so for raw idents, this will always be `None`.

Could be more optimal I guess, but it's a rare enough case that it wouldn't make much difference.